### PR TITLE
[image-manipulator] add doc comments in source

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-1761

To utilize the auto generation feature for API section on the docs website the package source code must contain the relevant documentation.

# How

This PR moves the current documentation of `ImageManipulator` API ( https://docs.expo.dev/versions/latest/sdk/imagemanipulator) with minor tweaks and additions to the package source code.

This PR also includes small tweak to the `APISectionInterface` component and  also updates the Snack example on the doc page.

# Test Plan

The changes and generated content have been tests running docs website on `localhost`.

# Preview

Screenshot 2021-08-07 at 22 16 35

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image-manipulator/src/ImageManipulator.types.ts [hunk 1]
```diff
@@ -1,31 +1,72 @@
...
+  /**
+   * Width of the image or video.
+   */
   width: number;
+  /**
+   * Height of the image or video.
+   */
   height: number;
+  /**
+   * it is included if the `base64` save option was truthy, and is a string containing the
```
Comment (tsapeta) [inline packages/expo-image-manipulator/src/ImageManipulator.types.ts:16]:
```suggestion
   * It is included if the `base64` save option was truthy, and is a string containing the
```

File: packages/expo-image-manipulator/src/ImageManipulator.types.ts [hunk 2]
```diff
@@ -1,31 +1,72 @@
...
+   * JPEG/PNG (depending on `format`) data of the image in Base64. Prepend that with `'data:image/xxx;base64,'`
+   * to get a data URI, which you can use as the source for an `Image` element for example
+   * (where `xxx` is `jpeg` or `png`).
+   */
   base64?: string;
 };
 
+// @needsAudit
+/**
+ * > If you specify only one value, the other will be calculated automatically to preserve image ratio.
```
Comment (tsapeta) [inline packages/expo-image-manipulator/src/ImageManipulator.types.ts:26]: I'm wondering if it shouldn't be added to `resize` comment instead, what do you think?

File: packages/expo-image-manipulator/src/ImageManipulator.types.ts [hunk 3]
```diff
@@ -1,31 +1,72 @@
...
+// @docsMissing
 export enum FlipType {
   Vertical = 'vertical',
   Horizontal = 'horizontal',
 }
 
+// @needsAudit
+/**
+ * > Only one flip per transformation is available. If you want to flip according to both axes then
+ * provide two separate transformations.
```
Comment (tsapeta) [inline packages/expo-image-manipulator/src/ImageManipulator.types.ts:56]: Same here, shouldn't it be a part of `flip` property documentation?

**Review by tsapeta — APPROVED** [review]

File: packages/expo-image-manipulator/src/ImageManipulator.types.ts (hunk 2 above)
Comment (Simek) [reply packages/expo-image-manipulator/src/ImageManipulator.types.ts:26]: @tsapeta It was my first take, however the note blocks do not render that well in the tables right now, and omitting the block results in blend-in of the message with the field description. 

But probably, as you suggested, it's the better place for this note, and we can improve the notes styling in tables during upcoming docs redesign. 😄

Comment (tsapeta) [reply packages/expo-image-manipulator/src/ImageManipulator.types.ts]: Honestly, it doesn't have to be a blockquote 😄 But I'm fine if only you'll remember to fix it in the future 😉

Comment (Simek) [reply packages/expo-image-manipulator/src/ImageManipulator.types.ts]: CSS changes are required to fix this, so they will apply automatically, but maybe it the right call to just skip the blockquote 😄

**Review by bbarthec — APPROVED** [review]
LGTM 👍 
Thank you for this one 💪 🥳

**Pull request merged**